core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala

```diff
@@ -168,40 +169,51 @@ class ContainerPool(
             info(this, s"Shutting down: Not getting container for ${action.fullyQualifiedName} with ${auth.uuid}")
             None
         } else {
-            try {
-                val myPos = nextPosition.next()
-                info(this, s"""Getting container for ${action.fullyQualifiedName} of kind ${action.exec.kind} with ${auth.uuid}:
-                              | myPos = $myPos
-                              | completed = ${completedPosition.cur}
-                              | slack = ${slack()}
-                              | startingCounter = ${startingCounter.cur}""".stripMargin)
-                val key = ActionContainerId(auth.uuid, action.fullyQualifiedName, action.rev)
-                getImpl(1, myPos, key, () => makeWhiskContainer(action, auth)) map {
-                    case (c, initResult) =>
-                        val cacheMsg = if (!initResult.isDefined) "(Cache Hit)" else "(Cache Miss)"
-                        (c.asInstanceOf[WhiskContainer], initResult)
-                }
-            } finally {
-                completedPosition.next()
+            val key = ActionContainerId(auth.uuid, action.fullyQualifiedName, action.rev)
+            val myPos = nextPosition.next()
+            info(this, s"""Getting container for ${action.fullyQualifiedName} of kind ${action.exec.kind} with ${auth.uuid}:
+                          | myPos = $myPos
+                          | completed = ${completedPosition.cur}
+                          | slack = ${slack()}
+                          | startingCounter = ${startingCounter.cur}""".stripMargin)
+            val conResult = Try(getContainer(1, myPos, key, () => makeWhiskContainer(action, auth)))
+            completedPosition.next()
+            conResult match {
+                case Success(Cold(con, initResult)) =>
+                    info(this, s"Obtained cold container ${con.containerId.id}")
+                    Some(con, Some(initResult))
+                case Success(Warm(con)) =>
+                    info(this, s"Obtained warm container ${con.containerId.id}")
+                    Some(con, None)
+                case Success(Error(str)) =>
```

Comment:
Expected error (docker misbehaving but we handled it) vs the future unexpectedly throwing an exception we had not thought of.   Different error modes.